src/celutil/logger.cpp

```diff
@@ -0,0 +1,81 @@
+#include <iostream>
+#include <fmt/ostream.h>
+#include <fmt/printf.h>
+#ifdef _MSC_VER
+#include <windows.h>
+#endif
+#include "logger.h"
+
+namespace celestia
+{
+namespace util
+{
+
+Logger* Logger::g_logger = nullptr;
+
+Logger* GetLogger()
+{
+    return Logger::g_logger;
+}
+
+Logger* CreateLogger(Level level)
+{
+    if (Logger::g_logger == nullptr)
+        Logger::g_logger = new Logger(level);
+    return Logger::g_logger;
+}
+
+void DestroyLogger()
+{
+    delete Logger::g_logger;
+}
+
+void Logger::vlog(Level level, fmt::string_view format, fmt::format_args args) const
+{
+#ifdef _MSC_VER
+    if (level == Level::Debug && IsDebuggerPresent())
+    {
+        OutputDebugStringA(fmt::vformat(format, args).c_str());
+        return;
+    }
+#endif
+
+    auto &stream = (level <= Level::Warning || level == Level::Debug) ? std::cerr : std::clog;
```

Comment:
It might be worth having the output streams as member variables in the logger (defaulting to clog and cerr), which would make this testable. Then we could move the code in the `#if 0` block at the end of this file into the test project.

Reply:
yeah, good idea